packages/flutter/lib/src/painting/strut_style.dart

```diff
@@ -0,0 +1,413 @@
+  /// [TextStyle.fontFamilyFallback], the font does not need to contain the
+  /// desired glyphs to match.
+  final String fontFamily;
+
+  /// The ordered list of font families to fall back on when a higher priority
+  /// font family cannot be found.
+  ///
+  /// The value provided in [fontFamily] will act as the preferred/first font
+  /// family that will be searched for, followed in order by the font families
+  /// in [fontFamilyFallback]. If all font families are exhausted and no match
+  /// was found, the default platform font family will be used instead. Unlike
+  /// [TextStyle.fontFamilyFallback], the font does not need to contain the
+  /// desired glyphs to match.
+  ///
+  /// When [fontFamily] is null or not provided, the first value in [fontFamilyFallback]
+  /// acts as the preferred/first font family. When neither is provided, then
+  /// the default platform font will be used. Providing and empty list or null
+  /// for this property is the same as omitting it.
+  ///
+  /// If the font is defined in a package, each font family in the list will be
+  /// prefixed with 'packages/package_name/' (e.g. 'packages/cool_fonts/Roboto').
+  /// The package name should be provided by the `package` argument in the
+  /// constructor.
+  List<String> get fontFamilyFallback => _package != null && _fontFamilyFallback != null ? _fontFamilyFallback.map((String str) => 'packages/$_package/$str').toList() : _fontFamilyFallback;
+  final List<String> _fontFamilyFallback;
+
+  // This is stored in order to prefix the fontFamilies in _fontFamilyFallback
+  // in the [fontFamilyFallback] getter.
+  final String _package;
+
+  /// The minimum size of glyphs (in logical pixels) to use when painting the text.
+  ///
+  /// During painting, the [fontSize] is multiplied by the current
+  /// `textScaleFactor` to let users make it easier to read text by increasing
+  /// its size.
+  ///
+  /// The default fontSize is `14` logical pixels.
+  final double fontSize;
+
+  /// The height of the line as a ratio of fontSize. This property does
+  /// not affect the leading height, which is controlled separately through
+  /// [leading].
+  ///
+  /// The default lineHeight is `1.0`.
+  final double lineHeight;
+
+  /// The typeface thickness to use when calculating the strut (e.g., bold).
+  ///
+  /// The default fontWeight is `w400`.
+  final FontWeight fontWeight;
+
+  /// The typeface variant to use when calculating the strut (e.g., italics).
+  ///
+  /// The default fontStyle is `normal`.
+  final FontStyle fontStyle;
+
+  /// The custom strut leading as a ratio of [fontSize].
+  ///
+  /// If this is null or negative, the font's leading will be used. Positive
+  /// values and zero specifies a custom leading, half of which will be applied
+  /// to the top of the line box and the other half to the bottom.
+  ///
+  /// The default leading is `-1`, which will use the font-specified leading.
+  final double leading;
```

Comment:
I think it is still very often useful to use the font's default leading. For example, if the user wants the strut to be the same as the rest of the text, if we do not provide an option to use default leading, then the user is unable to match the strut to the rest of the text as we do not (yet) provide direct access to metrics such as the leading. Normally, when the user provides a custom leading, the are trying to accomplish something specific. Otherwise, the default leading is likely what they wish to use for the majority of cases.

This API does not provide an easy way to make something 50% the size of the default leading. One (confusing) possibility is to make negative values a multiplier on the default leading. For example, `leading: -0.5` would indicate 50% of the default leading and we simply drop the sign.